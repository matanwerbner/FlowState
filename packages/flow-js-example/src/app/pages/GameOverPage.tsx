import { Button } from '@mui/material';
import { FlexBox } from '../components/FlexBox';
import { useGameStore } from '../store';

// shows "game over" message, and number of correct and incorrect responses.
export const GameOverPage = () => {
  const { context, triggerEvent } = useGameStore();
  return (
    <FlexBox>
      <h3>Game over!</h3>
      <div>Correct answers: {context.correctAnswers}</div>
      <div>Incorrect answers: {context.incorrectAnswers}</div>
      <Button
        onClick={() => {
          triggerEvent('playAgain');
        }}
      >
        Play again
      </Button>
    </FlexBox>
  );
};
