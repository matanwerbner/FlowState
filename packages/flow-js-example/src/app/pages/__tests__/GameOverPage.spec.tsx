import { render } from '@testing-library/react';
import { GameOverPage } from '../GameOverPage';

describe('GameOverPage tests', () => {
  it('should render', () => {
    const { baseElement } = render(<GameOverPage />);
    expect(baseElement).toMatchSnapshot();
  });
});
