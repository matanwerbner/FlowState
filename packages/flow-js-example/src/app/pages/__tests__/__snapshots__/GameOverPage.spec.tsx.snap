// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverPage tests should render 1`] = `
<body>
  <div>
    <div
      class="FlexBox__StyledBox-txfyk3-0 lgdOeB MuiBox-root css-329zn9"
    >
      <h3>
        Game over!
      </h3>
      <div>
        Correct answers: 
        0
      </div>
      <div>
        Incorrect answers: 
        0
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Play again
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
